content/blog/2018/01/2018-01-08-moving-from-buddybuild-for-android.adoc

```diff
@@ -0,0 +1,155 @@
+
+[.boxshadow]
+image:/images/post-images/2018-01-08/blue-ocean-screenshot.png[alt="Blue Ocean build screenshot",role=center]
+
+=== Building Android Apps
+To build an Android app, you need the Java development tools (JDK), which Jenkins can link:https://wiki.jenkins.io/display/JENKINS/Tool+Auto-Installation[automatically install] for you, plus the Android SDK, which you can also install on individual build agents using a link:https://wiki.jenkins.io/display/JENKINS/Tool+Auto-Installation[tool installer], or you can use a link:https://docker.com/[Docker container] with the Android SDK Tools preinstalled, for example.
+
+Then, you can use your SCM plugin of choice to fetch your source code, and build the app using the Android Gradle Plugin via the link:https://docs.gradle.org/4.4.1/userguide/gradle_wrapper.html[Gradle Wrapper] — in most cases this is as simple as running `./gradlew assembleDebug`.
+
+Once your app has been built and packaged into a `.apk` file, you can use the link:/doc/pipeline/steps/workflow-basic-steps/#code-archive-code-archive-artifacts[`archiveArtifacts`] build step, storing the APK, enabling colleagues to download APKs directly from Jenkins, so that they can try out the latest build.
+
+=== Testing Android Apps
+The Android SDK supports two types of test: unit tests, which run on the JVM, and instrumentation tests, which have to run on an Android device or emulator.  Both types of test can be executed using Jenkins and, since the Android Gradle Plugin writes the test results to disk in JUnit XML format, the plugin:junit[JUnit Plugin] for Jenkins can be used to parse the results, enabling you see a test report, and to be notified of test failures.
+
+Compiling and executing the unit tests for your app is as simple as adding another build step which runs `./gradlew testDebugUnitTest`.
+
+Similarly, instrumentation tests can be compiled and executed via the `connectedDebugAndroidTest` task in Gradle.  However, before you do this, you should ensure that an Android device is connected to your Jenkins build agent, or you can make use of the plugin:android-emulator[Android Emulator Plugin] to automatically download, create, and start an emulator for you during a build.  There are also plugins for cloud testing services such as link:https://github.com/jenkinsci/aws-device-farm-plugin[AWS Device Farm].
+
+Once you have finished executing the tests, you can use the link:/doc/pipeline/steps/junit/#code-junit-code-archive-junit-formatted-test-results[`junit`] step to analyse the results: `junit '**/TEST-*.xml'`.
```

Comment:
GitHub renders `*/TEST` formatted, may need `++` instead of backticks.

Reply:
`make run` shows it renders as expected 👍